azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_base.py

```diff
@@ -1028,6 +1029,17 @@ def _handle_pipeline_event(self, event):
         else:
             self.send_event_up(event)
 
+    def _always_try_reconnecting(self, error):
```

Comment:
>_always_try_reconnecting(self, error): [](start = 8, length = 38)

I'd like a little text documentation explaining this logic because it's not clear on it's face why this is necessary